USca/USca-Server/Tags/TagWorker.cs

```diff
@@ -118,10 +124,32 @@ private void UpdateTag()
                     // The tag points to a measure that doesn't exist.
                     return;
                 }
+                UpdateTagDataFrom(measure);
                 SendData(measure);
                 CheckAlarms(measure);
             }
 
+            private void UpdateTagDataFrom(Measure measure)
+            {
+                // Update the tag's current value to the one in the measure.
+                if (measure.Value != Tag.Value)
+                {
+                    using (var db = new ServerDbContext())
+                    {
+                        var tag = db.Tags.Find(Tag.Id);
+                        if (tag != null)
+                        {
+                            tag.Value = measure.Value;
+                            Tag.Value = measure.Value; // Do we need this?
+                            db.SaveChanges();
+                        }
+                    }
+                }
+
+                // Log the new value.
+                _tagLogService.AddFrom(Tag, measure.Timestamp);
+            }
```

Comment:
Is this necessary? Technically only input tags have a value, so for output tags we only need to add a `TagLog`, and not also update the `Tag` itself, neither the thread's local `Tag` nor the one we retrieve from and persist to the database.
Aside from the business logic concern, since this PR we are making a considerably larger number of database operations (my HDD is constantly churning). It's probably a good time to start looking at optimizations. Aside from this change, two general ones come to mind:
- Use in-memory databse
- Don't store `Measures` in database, but perhaps in a `Dictionary` in `TagWorker`? Writing a new `Measure` every 100ms (that's how often the RTU updates the readings now) to the database is expensive.

Reply:
The two methods below, `SendData` and `CheckAlarms`, use `Tag.Value`, which needs to be clamped. In other words, measure value != tag value. On the other hand, you're right that we don't actually have to save it to the database, since the `Tag` object is shared within the `TagThreadWrapper`, so the code can be simplified.

As for the first bullet point, we should consider _not_ erasing the database on every run. 

As for the second one: the RTU updates the readings every 100ms, but the data itself is sent every 250ms. Also it's written in a batch so it's 1 `SaveChanges` every 250ms. If you want to increase the period between two writes, go ahead.
I don't think that's a big cost on the I/O. It's the TagLogs that are problematic, because it's 1 I/O per tag, which is 4*N individual writes per second. I've batched the writes.

[2ae9460](https://github.com/magley/untitled-scada-project/pull/25/commits/2ae9460cf2ee20ad0a53e9f35a986c2f1d4efb8c)